worker/caas/manifold.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caas
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/worker.v1"
+
+	"github.com/juju/juju/api/agent"
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/caas"
+	"github.com/juju/juju/worker/dependency"
+)
+
+// ManifoldConfig describes the resources used by a Tracker.
+type ManifoldConfig struct {
+	APICallerName string
+	NewBrokerFunc caas.NewBrokerFunc
+}
+
+// Manifold returns a Manifold that encapsulates a *Tracker and exposes it as
+// an caas,Broker resource.
+func Manifold(config ManifoldConfig) dependency.Manifold {
+	manifold := dependency.Manifold{
+		Inputs: []string{
+			config.APICallerName,
+		},
+		Output: manifoldOutput,
+		Start: func(context dependency.Context) (worker.Worker, error) {
+			var apiCaller base.APICaller
+			if err := context.Get(config.APICallerName, &apiCaller); err != nil {
+				return nil, errors.Trace(err)
+			}
+			apiSt, err := agent.NewState(apiCaller)
```

Comment:
TODO(caas) introduce a caasbroker API

(this worker should not be using api/agent)

Reply:
Yeah the existing api had all the needed method(s)
The environ tracker has the same issue.